### PR TITLE
Mo 87 - Notify SPOs and POMs when no LDU information is available shortly before handover start date 

### PR DESCRIPTION
**Pull request by theKHutDeveloper**
This means that we have no way of notifying the LDU that they will be responsible for the offender. We alert the SPO and POM in these cases so that they can take necessary action outside of the application to deal with this circumstance.

There are 2 SPO views that will display this alert, allocations/new and allocations/show pages. Both views are quite similar so I am only displaying the allocations/new page in this PR.

### - **Allocate a Prison Offender Manager - top of the page**

Screen Shot 2020-11-20 at 15 52 44

### - **Allocate a Prison Offender Manager - lower part of the page**

Screen Shot 2020-11-20 at 15 53 01

**-------------------------------------------------------------------------------------------**

There is one POM view that will display this alert on the prisoners/show page

### - **Prisoner Information - top of the page**
Screen Shot 2020-11-20 at 15 54 49

### - **Prisoner Information - bottom of the page**
Screen Shot 2020-11-20 at 15 55 00

### CODE REVIEW

File: app/helpers/offender_helper.rb [hunk 1]
```diff
@@ -30,4 +30,18 @@ def event_type(event)
       type + 'allocated'
     end
   end
+
+  def approaching_handover_without_com?(offender)
+    return false unless offender.sentenced?
```
Comment (starswan) [inline app/helpers/offender_helper.rb:35]: I think this method might be nicer as a 'traditional if statement'. If I'm reading it correctly, it can return false even when com_name is not present. (if there's no email address to talk to) which seems a little counter-intuitive given the name of the method. 
I also wonder if it would be better if it returned the positive case rather than the negative one? (unsure)

File: app/views/shared/_no_com_alert.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="govuk-error-summary">
+  <h2 class="govuk-error-summary__title" id="error-summary-title">
+    You must contact the community probation office
+  </h2>
+  <div class="govuk-error-summary__body">
+    <ul class="govuk-list govuk-error-summary__list">
+      <li id="com_warning"><%= link_to I18n.t('views.handover_alert.no_com', date: format_date(@prisoner.handover_start_date)), "#com-error", data: { turbolinks: false } %></li>
+      <li id="ldu_warning">
+        <% if @prisoner.ldu.try(:name).blank? %>
```
Comment (starswan) [inline app/views/shared/_no_com_alert.html.erb:9]: This looks like this shoudl be encapsulated as @prisoner.has_ldu?

File: app/helpers/offender_helper.rb (hunk 1 above)
Comment (starswan) [reply app/helpers/offender_helper.rb:35]: Maybe this should be split into 2 for readability. approaching_handover_without_com? sounds like a question, as does offender.has_ldu_email? Then the checks can be more explicit i.e. if approaching_handover_without_com? && offender.ldu_email_missing?

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -50,3 +50,13 @@ a:visited {
  .govuk-header__navigation-item a {
      color: white !important;
  }
+
```
Comment (starswan) [inline app/assets/stylesheets/application.scss:53]: Slightly surprised that this isn't available already - maybe a comment would be appropriate?

File: app/helpers/offender_helper.rb [hunk 4]
```diff
@@ -30,4 +30,18 @@ def event_type(event)
       type + 'allocated'
     end
   end
+
+  def approaching_handover_without_com?(offender)
```
Comment (ollietreend) [inline app/helpers/offender_helper.rb:34]: I'm wondering if this might be a more descriptive name for this method:

```suggestion
  def needs_com_but_ldu_is_uncontactable?(offender)
```

It's a bit longer, but hopefully more descriptive. Not sure if it's too wordy though?

No need to use this, though. Naming things is hard 😓

**Review by ollietreend — APPROVED** [review]
This looks good 🎉

I've left one comment about the helper method name – but this is just an opinion about naming, so no need to use it!
